src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -206,24 +260,105 @@ public static LogicalLocation CreateLogicalLocation(LogicalLocationVersionOne v1
             return logicalLocation;
         }
 
-        public static IList<Notification> CreateNotificationsList(IEnumerable<NotificationVersionOne> v1Notifications)
+        internal LogicalLocation CreateLogicalLocation(string fullyQualifiedLogicalName, string parentKey = null, string decoratedName = null)
         {
-            List<Notification> notifications = null;
+            var logicalLocation = new LogicalLocation
+            {
+                DecoratedName = decoratedName,
+                FullyQualifiedName = fullyQualifiedLogicalName,
+                Name = GetLogicalLocationName(fullyQualifiedLogicalName),
+                ParentKey = parentKey
+            };
+
+            return logicalLocation;
+        }
 
-            if (v1Notifications != null)
+        internal string AddLogicalLocation(LogicalLocation logicalLocation)
+        {
+            if (_currentRun.LogicalLocations == null)
             {
-                notifications = new List<Notification>();
+                _currentRun.LogicalLocations = new Dictionary<string, LogicalLocation>();
+            }
+
+            string fullyQualifiedName = logicalLocation.FullyQualifiedName;
+            string logicalLocationKey = logicalLocation.FullyQualifiedName;
+            int disambiguator = 0;
 
-                foreach (NotificationVersionOne notification in v1Notifications)
+            // We need to do the comparison here too
```

Comment:
What does that mean. Apparently you're doing it somewhere else, but it's not immediately apparent where. Also, it's not clear _why_ you need to do the comparison in two places.

Reply:
Sorry, that was a note for me, meant to remove it.